### PR TITLE
Side menu UI

### PR DESCRIPTION
**Pull request by veverlymae**
# 📲 What
- UI/UX implementation for SideMenu Page

# 🤔 Why
- N/A

# 🛠 How
- 

# ✅ Acceptance criteria
- N/A

# ⏰ TODO
- 

# 🔗 References
https://xd.adobe.com/view/88019197-b52c-4c03-a717-e10207152dd7-dd52/screen/3054fe88-5fad-4afc-b0a2-de4b1958f191/specs/
Screen Shot 2021-06-03 at 7 26 06 PM

### CODE REVIEW

**Review by ljylaya — APPROVED** [review]

File: Kwartracker/Scenes/ContentView.swift [hunk 1]
```diff
@@ -8,9 +8,28 @@
...
+    @State var showSidebar: Bool = false
+        
     var body: some View {
-        Text("Hello, world!")
-            .padding()
+        
+        //https://blckbirds.com/post/side-menu-hamburger-menu-in-swiftui/
+        let drag = DragGesture()
+            .onEnded {
+                if $0.translation.width < -100 {
```
Comment (1jacobb1) [inline Kwartracker/Scenes/ContentView.swift:19]: Create a variable to store `-100`, avoid using magic numbers.

File: Kwartracker/Scenes/ContentView.swift [hunk 2]
```diff
@@ -8,9 +8,28 @@
...
+                    withAnimation {
+                        self.showSidebar = false
+                    }
+                }
+            }
+        
+        return SideBarStack(sidebarWidth: 300, showSidebar: $showSidebar) {
```
Comment (1jacobb1) [inline Kwartracker/Scenes/ContentView.swift:26]: Store `300` to a variable, avoid magic number.

File: Kwartracker/Scenes/Reusable/MenuView.swift [hunk 3]
```diff
@@ -0,0 +1,20 @@
+//
+//  MenuView.swift
+//  Kwartracker
+//
+//  Created by Veverly Mae Ricaza on 5/24/21.
+//
+
+import SwiftUI
+
+struct MenuView: View {
```
Comment (1jacobb1) [inline Kwartracker/Scenes/Reusable/MenuView.swift:10]: It seems like this view is not being use?

File: Kwartracker/Scenes/Reusable/ProfileView.swift [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+//
+
+import SwiftUI
+
+struct ProfileView: View {
+    var body: some View {
+        HStack {
+            Image(uiImage: Asset.Images.icSample.image)
+                .resizable()
+                .frame(width: 40, height: 40, alignment: .center)
```
Comment (1jacobb1) [inline Kwartracker/Scenes/Reusable/ProfileView.swift:15]: Avoid using magic numbers

File: Kwartracker/Scenes/Reusable/SideBarStack.swift [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+                        } else {
+                            Color.clear
+                            .opacity(showSidebar ? 0 : 0)
+                            .onTapGesture {
+                                self.showSidebar = false
+                            }
+                        }
+                    }
+                )
+                .offset(x: showSidebar ? sidebarWidth : 0, y: 0)
```
Comment (1jacobb1) [inline Kwartracker/Scenes/Reusable/SideBarStack.swift:53]: Avoid using magic numbers

File: Kwartracker/Scenes/Side Menu/SideMenuView.swift [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+                        .resizable()
+                        .frame(width: 15, height: 15, alignment: .center)
+                    Text(L10n.SideMenu.Label.logout)
+                        .font(.system(size: 16))
+                        .foregroundColor(Color(.white))
+                }
+            }
+            .padding([.leading, .top], 25)
+        }
+        .padding([.top, .bottom], 30)
```
Comment (1jacobb1) [inline Kwartracker/Scenes/Side Menu/SideMenuView.swift:46]: Avoid using magic numbers

File: Kwartracker/Scenes/Side Menu/SideMenuView.swift [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+            } else {
+                didTap = true
+            }
+        }) {
+            HStack {
+                Image(uiImage: sideMenu.image)
+                    .frame(width: 16, height: 14, alignment: .leading)
+                    .padding(.leading, 10)
+                Text(sideMenu.text)
+                    .font(.system(size: 16))
```
Comment (1jacobb1) [inline Kwartracker/Scenes/Side Menu/SideMenuView.swift:72]: Avoid using magic numbers

**Review by 1jacobb1 — CHANGES_REQUESTED** [review]

File: Kwartracker/Scenes/Reusable/MenuView.swift (hunk 3 above)
Comment (veverlymae) [reply Kwartracker/Scenes/Reusable/MenuView.swift:10]: Deleted already

**Review by 1jacobb1 — APPROVED** [review]

**Pull request merged**